### DISCUSSION TITLE
Is there a way to process a JSON file larger than memory?

### DISCUSSION THREAD

**Discussion by DonPug** [nlohmann/json#3150] [Q&A] [answered]
Hey,

Based on an another discussion (linked below) memory is a hard limit for processing JSON files.
https://github.com/nlohmann/json/discussions/2833

Is there any way to let's say process a 20GB JSON file with only 4GB of memory?
If yes, what would be the best way to do it?

**Comment by nlohmann** [comment, marked as answer]
Depends on what you want to process. There is a SAX interface that allows to generate events during parsing that you can react to. So if it's not about building a DOM, but rather doing something with less storage requirements, 20 GB should not be an issue.

**Reply by DonPug** [reply]
It's just about creating an another JSON, the first one contains inputs, which will be used for calculations, and the output JSON will store the results. So I should use the SAX interface and should be okay with the low memory?

**Reply by nlohmann** [reply]
Yes, the SAX interface is the way to go. Reach out if you need further assistance!

**Reply by DonPug** [reply]
Great! Thanks for the quick answers!
